learn-array-methods-by-building-a-statistics-calculator/script.js

```diff
@@ -0,0 +1,42 @@
+const getMean = (array) => {
+  // teach the reduce method
+  const mean = array.reduce((acc, el) => acc + el, 0) / array.length;
+  return mean;
+};
+
+const getMedian = (array) => {
+  const median = array[Math.floor((array.length - 1) / 2)];
+  return median;
+};
+
+const getMode = (array) => {
+  const counts = {};
+  // teach the forEach method??
+  array.forEach((el) => {
+    counts[el] = (counts[el] || 0) + 1;
+  });
+  const highest = Object.keys(counts).sort(
+    (a, b) => counts[b] - counts[a]
+  )[0][0];
```

Comment:
I think this is the first time we introduce method chaining.
So we do want to make sure to explain it here. 
Also, this might be the first time we introduce `Object.keys`

Reply:
We could also chain the `map` and `filter` and explain chaining there, so that we can explain `Object.keys` here.

Reply:
> We could also chain the `map` and `filter` and explain chaining there, so that we can explain `Object.keys` here.

That sounds like a great idea.